### ISSUE TITLE
Overloading Tag and Taggins

### ISSUE THREAD

**Issue by arjes** [laserlemon/vestal_versions#26]
Using Thinking Shpinx, and Formtastic with vestal_versions they bring up some interesting errors... Namely:

"undefined method `reflect_on_association' for VestalVersions::Tagging:Module"

I assume this is because these addons are trying to talk to VestalVersions and its not responding as it expects?

**Comment by arjes** [comment]
Updating:

Alright so everything is happy, as long as thinking sphinx is not indexing on the Tag IDs... I assume this is becasue vestal_versions is adding its own associations for tags.

This is a fairly common association (I use Tagging and Tag), is there any way to rename or scope this out?

**Comment by arjes** [comment]
Oh, and the model that I can't index on the tag_ids is NOT versioned.,

additionally even when this index is commented out, formtastic returns the same error when I try to get it to display my :tags association.

**Comment by dfurber** [comment]
I have the same problem but doesn't seem to have anything to do with Thinking Sphinx. Seems to be a generic name conflict with acts_as_taggable_on_steroids.

**Comment by bishopandco** [comment]
Not TS here. Just acts_as_taggable_on and getting the error as well.

**Comment by dfurber** [comment]
Sorry about the reference to Thinking Sphinx. The problem is that vestal versions has a Tag class, and acts_as_taggable_on also defines a Tag model. I fixed it by forking the gem and namespacing the vestal versions tag to be VestalVersions::Tag, and letting acts as taggable have Tag, since it's an active record model.

**Comment by bishopandco** [comment]
Yeah I did it much dirtier. I forked and renamed all things tag to versionable_tag :)

**Comment by stuffmc** [comment]
One thing I don't understand... This is specific to one of the latest versions, right? With 0.8.3 it was working fine. It should have been fixed in 1.0 with the namespace, but it doesn't here :( Right now I'm using 0.8.3 specifically because 1.0.2 crashes.

**Comment by stuffmc** [comment]
Nothing new here? Is vestal_version ever gonna be updated? With the last commit being almost 10 months old I guess I'll look for another option (if there is) on the next project.

**Comment by bishopandco** [comment]
it still works and does everything you need it to out of the box.

**Comment by stuffmc** [comment]
0.8.3 does, but as of 1.0 there's this Tagging bug (9 month old) which makes it not usable. So I am to use a version which is 1 year old. But yeah, it works and it's great. It would just be awesome if this bug was fixed so I can use 1.0..... 3 :)

**Comment by adamcooper** [comment]
We are using this plugin and planning to maintain our fork of it and we've been pretty active on trying to keep it up to date and adding new features.  

It seems there are a couple of issues that are residing in different forks. (initial version, and tagging).  If people are interested I can pull them all together in one fork.

You can see the fork here: 
http://github.com/adamcooper/vestal_versions

Thanks,
Adam

**Comment by bishopandco** [comment]
Our fork is a hack. I wouldn't pull ours :)

**Comment by adamcooper** [comment]
Hehe.  It's not any more hacky than the other fixes.  ;)

I ended up pulling in another branch though because it seemed less intrusive.  Mostly, it didn't require a schema change as the conflict was only at the module level.
